rust/src/lib/plugin.rs

```diff
@@ -0,0 +1,16 @@
+use crate::{nm::version, NmstateError};
+
+use serde::Serialize;
+
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize)]
+pub struct Plugin {
+    name: String,
+    version: String,
+}
+
+pub(crate) fn list_plugins() -> Result<Vec<Plugin>, NmstateError> {
+    Ok(vec![Plugin {
+        name: "NetworkManager".to_string(),
+        version: version::nm_version().unwrap(),
+    }])
+}
```

Comment:
I am not sure if this makes sense.. nispor and ovsdb should be there too.. what do you think @cathay4t?

Reply:
nispor and ovsdb version is not required as rust is doing pull-in dependency, not runtime.